src/main/java/sc/fiji/bdvpg/bdv/projector/AccumulateMixedProjectorARGB.java

```diff
@@ -136,9 +136,9 @@ else if ( projectionModes[sourceIndex].equals( Projection.PROJECTION_MODE_AVG ))
 			if ( source instanceof TransformedSource )
 				source = ( ( TransformedSource ) source ).getWrappedSource();
 
-			SourceAndConverter sac = getSourceAndConverter( sacs, source );
+			SourceAndConverter sac = SourceAndConverterServices.getSourceAndConverterService().getSourceAndConverterFromSource( source );
 
-			final String projectionMode = (String) SacServices.getSacService().getMetadata( sac, PROJECTION_MODE );
+			final String projectionMode = (String) SourceAndConverterServices.getSourceAndConverterService().getMetadata( sac, PROJECTION_MODE );
```

Comment:
Hi @tpietzsch,
We want to implement `SourceAndConverter`-dependent projection modes. For this to work, we currently look within the `AccumulateProjector` in a global map which projection mode is associated with which `SourceAndConverter`. Now, unfortunately the `AccumulateProjector` only knows about a `List< Source >` and not a `List< SourceAndConverter >`. This is a challenge for us, because one java `Source` object could correspond to multiple `SourceAndConverter`. Thus we do not know how to properly do the mapping. Do you have some advise?


@NicoKiaru

Reply:
@NicoKiaru

As discussed, I managed to inject a `bdvHandle` into the `AccumulateMixedProjectorARGB` (https://github.com/bigdataviewer/bigdataviewer-playground/pull/73/commits/df97e5db522db2c3c087546e09255dfee2a7e890)

There is now a function
`getProjectionModes( BdvHandle bdvHandle, ArrayList< Source< ? > > sources )`
from which you could continue working on, as discussed.

Thanks!

Reply:
@tischi @NicoKiaru I'll change it, so that `createAccumulateProjector()` has a
`List<SourceAndConverter<?>> sources` parameter instead of `List<Source<?>>`.

Reply:
Thanks @tpietzsch !
In the new Improved UI branch or in the old one ?

Reply:
>  so that createAccumulateProjector() has a
List<SourceAndConverter<?>> sources parameter instead of List<Source<?>>.

awesome, that was in fact my dream :-)

Reply:
Thanks for your input!

Reply:
Resolved in old version. Need to check on the branch for the new ui how to solve it in more simple manner

Reply:
@NicoKiaru @tischi the new version would be in this branch
https://github.com/bigdataviewer/bigdataviewer-core/tree/projector-factory
(on top of the new ui branch)
Do you think this is useful?

Reply:
@tpietzsch That would be wonderful! 
Just to make sure: can we rely on the `SourceAndConverter` objects in the list being identical to the ones that we add using `bdvHandle.getViewerPanel().addSource( sac )`? Or could there be some wrapping happening inside `bdv`?

Reply:
Yes, you can rely on that they are `.equals()`.

At the moment, they are even the same object `==`, but I would that might change at some point, so better to use `.equals()`.

Reply:
of course, looks perfect. We have several workarounds at the moment to make this work for the old version, but that would be reeally nice to remove them.

Reply:
Ok, perfect, then I'll just merge it to the "new ui" branch and then that should be merged to master pretty soon